impeller/geometry/path.h

```diff
@@ -90,10 +90,29 @@ class Path {
 
   /// One or more contours represented as a series of points and indices in
   /// the point vector representing the start of a new contour.
+  ///
+  /// Polylines are ephemeral and meant to be used by the tessellator. They do
+  /// not allocate their own point vectors to allow for optimizations around
+  /// allocation and reuse of arenas.
   struct Polyline {
+    /// The signature of a method called when it is safe to reclaim the point
+    /// buffer provided to the constructor of this object.
+    using PointBufferPointer = std::unique_ptr<std::vector<Point>>;
+    using ReclaimPointBuffer = std::function<void(PointBufferPointer)>;
+
+    /// The buffer will be cleared and returned at the destruction of this
+    /// polyline.
+    Polyline(PointBufferPointer point_buffer, ReclaimPointBuffer reclaim);
```

Comment:
This is a bit awkward but given c++'s lack of lifetimes I think this is an improvement and maybe the best we can do.  I think you should move the polyline tests to a path_unittests.cc.  Having the tests in a nonstandard location makes it hard to verify coverage.

Reply:
I think if we want to break up geometry_unittests.cc maybe let's do it in a separate PR.

Reply:
A followup PR sounds good to me.  It's a bit easier to review this one if the tests are in the proper place but I can understand wanting to separate refactoring and functional changes.